### PR TITLE
Dev dust

### PR DESCRIPTION
**Pull request by helenarichie**

### CODE REVIEW

File: src/gravity/gravity_functions.cpp [hunk 1]
```diff
@@ -156,7 +159,18 @@ void Grid3D::set_dt_Gravity(){
 
   #if defined( AVERAGE_SLOW_CELLS) && !defined( PARTICLES )
   //Set the min_delta_t for averaging a slow cell ( for now the min_dt_slow is set to a large value, change this with your condition )
-  min_dt_slow = H.dt / C_cfl * 100 ;
+  // min_dt_slow = H.dt / C_cfl * 100 ;
+  printf("/nhowdy!!/n");
```
Comment (alwinm) [inline src/gravity/gravity_functions.cpp:163]: extraneous printfs

File: src/global/global.h [hunk 2]
```diff
@@ -54,7 +54,7 @@ typedef double Real;
 #define LOG_FILE_NAME "run_output.log"
 
 //Conserved Floor Values
-#define TEMP_FLOOR 1e-3 // in Kelvin
+#define TEMP_FLOOR 10 // in Kelvin
```
Comment (alwinm) [inline src/global/global.h:57]: should this change be wrapped in ifdef DUST so as to not change temp floor for others? not really sure, myself.

File: src/grid/cuda_boundaries.cu [hunk 3]
```diff
@@ -357,6 +357,58 @@ __device__ int FindIndex(int ig, int nx, int flag, int face, int n_ghost, Real *
...
+
+  if (xid <= n_ghost && xid < nx && yid < ny && zid < nz) {
+    // set conserved variables
+    c_device[gid] = d_0;
+    c_device[gid+1*n_cells] = vx*d_0;
+    c_device[gid+2*n_cells] = vy*d_0;
+    c_device[gid+3*n_cells] = vz*d_0;
+    c_device[gid+4*n_cells] = P_0/(gamma-1.0) + 0.5*d_0*(vx*vx + vy*vy + vz*vz);
+  }
+  __syncthreads();
```
Comment (alwinm) [inline src/grid/cuda_boundaries.cu:408]: is syncthreads necessary here?

File: src/grid/grid3D.cpp [hunk 4]
```diff
@@ -136,7 +140,8 @@ void Grid3D::Initialize(struct parameters *P)
   C_cfl = 0.3;
   
   #ifdef AVERAGE_SLOW_CELLS
-  H.min_dt_slow = 1e-100; //Initialize the minumum dt to a tiny number
+  // H.min_dt_slow = 1e-10; //Initialize the minumum dt to a tiny number
+  H.min_dt_slow = 1e-2;
```
Comment (alwinm) [inline src/grid/grid3D.cpp:144]: should this be wrapped in ifdef DUST as well? not really sure how changing min_dt_slow affects other parts of code

File: src/grid/initial_conditions.cpp [hunk 5]
```diff
@@ -587,7 +591,7 @@ void Grid3D::KH()
   Real vx, vy, vz;
   Real d1, d2, v1, v2, P, A;
 
-  d1 = 2.0;
+  d1 = 200.0;
```
Comment (alwinm) [inline src/grid/initial_conditions.cpp:594]: is this change to KH intentional?

File: src/hydro/hydro_cuda.cu [hunk 6]
```diff
@@ -566,6 +566,21 @@ __global__ void Calc_dt_3D(Real *dev_conserved, Real *dev_dti, Real gamma, int n
         max_dti = fmax(max_dti,hydroInverseCrossingTime(E, d, d_inv, vx, vy, vz, dx, dy, dz, gamma));
       #endif  //MHD
 
+      Real P  = (E - 0.5*d*(vx*vx + vy*vy + vz*vz)) * (gamma - 1.0);
+      Real cs = sqrt(d_inv * gamma * P);
+      Real n = d*DENSITY_UNIT/(0.6*MP);
+      Real T = hydro_utilities::Calc_Temp(P, n);
+
+      if (max_dti > 1) {
+        printf("\nmax_dti: %e\n", max_dti);
```
Comment (alwinm) [inline src/hydro/hydro_cuda.cu:575]: are these temporary debug printfs or intended permanently?

**Comment by alwinm** [comment]
Great PR overall, I've added comments where changes might impact non-DUST behavior. 

Looks like the meat of the changes are the Dust kernel and wind tunnel boundaries/init, which I did not look at too closely.

File: src/grid/cuda_boundaries.cu (hunk 3 above)
Comment (twhite-cray) [reply src/grid/cuda_boundaries.cu:408]: No, not needed at the end of a kernel.

**Pull request closed without merging**